core/include/detray/propagator/rk_stepper.hpp

```diff
@@ -68,6 +73,8 @@ class rk_stepper final : public base_stepper<track_t, constraint_t, policy_t> {
             array_t<scalar, 4> k_qop;
         } _step_data;
 
+        const magnetic_field_t _magnetic_field;
```

Comment:
Is this a good idea? A stepper state exists for every track as part of the propagator state, which would mean we end up with thousands of b-field copies if we don't pass this by pointer. Also, the b-field [pointer] in the stepper instead of the e.g. the propagator state means, we get a b-field data member in the line stepper, which does not seem right in terms of inheritance. For the navigator we have the tracking geometry as a pointer already. I know its bad design to have to guarantee the lifetime of the data outside the class that uses it, but I don't see a good way to do it differently?

Reply:
I agree. I changed it to a pointer